main.go

```diff
@@ -732,9 +734,37 @@ func executeLinter(state *linterState, args []string) error {
 			state.Name)
 	}
 
-	processOutput(state, buf.Bytes())
+	cmdOutput, err := preprocessLinterPanic(state.Name, buf, err)
+	processOutput(state, cmdOutput)
 	debug("%s linter took %s", state.Name, time.Since(start))
-	return nil
+	return err
+}
+
+const panicExitStatus = 2
+
+func preprocessLinterPanic(name string, buf *bytes.Buffer, processError error) ([]byte, error) {
+	if exitStatus(processError) == panicExitStatus {
```

Comment:
> if the function returns then all the output from the linter would be discarded and gometalinter will think it panicked.

The error is returned as the last line, it's not an early return, so this is not true. All the linter output will still be processed and printed. At the very end the user will see an extra line warning that the linter may have panicked. They will always see all of the linter output.